File: sdk/src/main/java/com/sygic/travel/sdk/contentProvider/api/LocaleInterceptor.java
package com.sygic.travel.sdk.contentProvider.api;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by michal.murin on 16.2.2017.
 */

public class LocaleInterceptor implements Interceptor{
	public static final String TO_INTERCEPT = "[toIntercept]";

	private String locale;

	public LocaleInterceptor(){
		updateLocale();
	}

	public void updateLocale(){
		locale = SupportedLanguages.getActualLocale();
	}

	@Override
	public Response intercept(Interceptor.Chain chain) throws IOException {
		Request original = chain.request();
		String url = original.url().toString();

		url = url.replace(TO_INTERCEPT, ApiClient.API_VERSION + "/" + locale);

		Request request = original.newBuilder()
			.url(url)
			.method(original.method(), original.body())
			.build();

		return chain.proceed(request);
	}
}

File: sdk/src/main/java/com/sygic/travel/sdk/api/interceptors/LocaleInterceptor.kt
package com.sygic.travel.sdk.api.interceptors

import com.sygic.travel.sdk.api.StApiConstants.API_VERSION
import com.sygic.travel.sdk.api.StApiConstants.VERSION_AND_LOCALE
import com.sygic.travel.sdk.api.SupportedLanguages

/**
 *
 * Implements [okhttp3.Interceptor], adds API version and locale code.
 */
internal class LocaleInterceptor : okhttp3.Interceptor {
    /**
     *
     * Device's current locale code
     */
    private var locale: String? = null

    init {
        updateLocale()
    }

    /**
     *
     * Updates [LocaleInterceptor.locale] to device's current locale code
     */
    fun updateLocale() {
        locale = SupportedLanguages.actualLocale
    }

    /**
     *
     * Modifies the original request by adding an **API version** and a **locale code**.
     */
    @Throws(java.io.IOException::class)
    override fun intercept(chain: okhttp3.Interceptor.Chain): okhttp3.Response {
        val original = chain.request()
        var url = original.url().toString()

        url = url.replace(VERSION_AND_LOCALE, API_VERSION + "/" + locale)

        val request = original.newBuilder()
                .url(url)
                .method(original.method(), original.body())
                .build()

        return chain.proceed(request)
    }
}
